From e5e32269621794eab533cd071d8a1f7d880d3eb5 Mon Sep 17 00:00:00 2001
From: Neil Cheng <Neil.Cheng@quantatw.com>
Date: Wed, 6 May 2026 10:46:41 +0800
Subject: [PATCH] ARM: dts: aspeed: rainiera6: configure tach channels for
 max31790

Follow Yosemite5 since rainiera6 use the same PDB.
Configure the TACH sensor channels in the MAX31790 fan controller node.
This functionality is not supported in the upstream device tree,
so the change is applied only in the Meta tree.

Signed-off-by: Neil Cheng <Neil.Cheng@quantatw.com>
---
 .../boot/dts/aspeed/aspeed-bmc-facebook-rainiera6.dts  | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/arch/arm/boot/dts/aspeed/aspeed-bmc-facebook-rainiera6.dts b/arch/arm/boot/dts/aspeed/aspeed-bmc-facebook-rainiera6.dts
index 19aa8a391..36d1c0373 100644
--- a/arch/arm/boot/dts/aspeed/aspeed-bmc-facebook-rainiera6.dts
+++ b/arch/arm/boot/dts/aspeed/aspeed-bmc-facebook-rainiera6.dts
@@ -518,6 +518,16 @@ psu@5c {
 	fan-controller@5e{
 		compatible = "maxim,max31790";
 		reg = <0x5e>;
+		#address-cells = <1>;
+		#size-cells = <0>;
+		channel@1 {
+			reg = <1>;
+			sensor-type = "TACH";
+		};
+		channel@2 {
+			reg = <2>;
+			sensor-type = "TACH";
+		};
 	};
 
 	/* I2C MUX for PWRPIC #1, #2, #11, #12 */
